tests/bsim/bluetooth/host/gatt/authorization/src/gatt_client_test.c

```diff
@@ -297,6 +297,8 @@ static void test_main(void)
 {
 	int err;
 
+	bt_conn_cb_register(&conn_callbacks);
+
```

Comment:
Explicitly register is required because? doesn't `BT_CONN_CB_DEFINE`  work? All samples use it, is there more to be fixed?

Reply:
> Explicitly register is required because? doesn't `BT_CONN_CB_DEFINE` work? All samples use it, is there more to be fixed?

The commit message mentions double registration, so I assume there was some place where the static registration caused problems. Admittedly the the commit message didn't really explain how/where this was happening, so I'm also curious to hear the answer to this.

Reply:
The commit message is misleading. The problem is not that anything got registered twice. It's that `gatt_client_test.c` and `gatt_server_test.c` are compiled in to the same image.

`gatt_client_test.c` is causing a problem:

`gatt_client_test.c:disconencted` does `bt_conn_unref(g_conn)`. `g_conn` is `NULL` in this situation, causing undefined behavior.

Note that `g_conn` is not set in `gatt_client_test.c:connected`.

When asserts are on, `bt_conn_unref(NULL)` panics.

This patch fixes the problem by only registering the problematic `gatt_client_test.c:disconencted` when running the test it was made for.

Reply:
Yes I'm sorry. I have to stop writing like the reader is me :)
Commit message updated, thanks for translating my cryptic message @alwa-nordic .

Reply:
It seems like CI will retrigger all builds and tests when only the commit message is updated. Noice.